### DISCUSSION TITLE
Is there a way of catching errors in icecast output ?

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#1894] [Q&A]
I'm running a script where an input.http source and icecast output are being dynamically created.

Situations occur where I get a error from both output.icecast and input.http . Is there a way of catching these errors ?

I have tried try/catch , but I don't think it was made for this ?

Thanks

**Comment by Russsgithub** [comment]
I worked around this by calling http.head on the input.http stream url and only created it if the response status code is below 400.

Icecast error - I send a very small file to the url port mount through process.run(ffmpeg ...... ) and used the status code of that response as above.

It's be nice to have error catching of sources, but this works,.

**Reply by toots** [reply]
Hi,

I think `http.head` is great for plain HTTP requests, that's the way the protocol is suppose to work.

Have you tried it with `output.icecast` as well? Recent icecast servers do support regular http `PUT` request so it's possible they also support head requests..

**Reply by Russsgithub** [reply]
@toots 
I'll try, thanks.

**Reply by Russsgithub** [reply]
I couldn't work out how to add credentials to http.head, if I sent a request to a possible mount I got 404 back every time. Though I did move to curl to get the header rather than ffmpeg which works fine and removed the need to push a small file out.
